### PR TITLE
Handle the repo listing feature through the apiv2

### PR DESCRIPTION
**Pull request by morucci**

### CODE REVIEW

File: haskell/src/Monocle/Backend/Queries.hs [hunk 1]
```diff
@@ -287,6 +288,32 @@ getProjectAgg query = do
...
+    -- Terms agg returns empty terms in a buckets
+    isNotEmptyTerm :: BH.TermsResult -> Bool
+    isNotEmptyTerm tr = getTermKey tr /= ""
+
+getRepos :: TenantM [Text]
+getRepos = do
+  results <- runTermAgg
+  pure $ getRepoFullName <$> trace (show results) results
+  where
+    query = fromMaybe (error "oops") $ Q.queryBH $ Q.load Nothing mempty Nothing "repo_regex: .*"
```
Comment (TristanCacqueray) [inline haskell/src/Monocle/Backend/Queries.hs:313]: If you'd like to define query using the search language, we could look into using a QuasiQuote, or make that `fromMaybe undefined . Q.queryBH . Q.load Nothing mempty Nothing` a utility function in the Query module.

File: haskell/src/Monocle/Backend/Queries.hs [hunk 2]
```diff
@@ -287,6 +288,32 @@ getProjectAgg query = do
...
+getTermKey BH.TermsResult {} = error "Unexpected match"
+
+getTermsAgg :: BH.Query -> Text -> TenantM [BH.TermsResult]
+getTermsAgg query onTerm = do
+  search <- aggSearch query aggs
+  pure $ filter isNotEmptyTerm $ unfilteredR search
+  where
+    aggs = BH.mkAggregations "singleTermAgg" $ BH.TermsAgg $ BH.mkTermsAggregation onTerm
+    toTermsResult bucket = BH.buckets bucket
+    unfilteredR search' = maybe [] toTermsResult (BH.toTerms "singleTermAgg" search')
```
Comment (TristanCacqueray) [inline haskell/src/Monocle/Backend/Queries.hs:303]:
```suggestion
    unfilteredR search' = maybe [] BH.buckets (BH.toTerms "singleTermAgg" search')
```

Comment (TristanCacqueray) [reply haskell/src/Monocle/Backend/Queries.hs:303]: And using eta-reduction: `unfilteredR = maybe [] BH.buckets . BH.toTerms "singleTermAgg"`

File: haskell/src/Monocle/Backend/Queries.hs [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications #-}
```
Comment (TristanCacqueray) [inline haskell/src/Monocle/Backend/Queries.hs:2]: Is there any TypeApplications, or is this a left-over?

File: haskell/src/Monocle/Backend/Queries.hs (hunk 1 above)
Comment (morucci) [reply haskell/src/Monocle/Backend/Queries.hs:313]: I'll use an helper function for the moment.

**Review by TristanCacqueray — APPROVED** [review]
LGTM, would you like to merge this now and do the UI in another PR?

**Comment by morucci** [comment]
Yes please merge now. I'll tackle the UI in another PR.

**Pull request merged**